### PR TITLE
NMS-8671 - handle SNMP error-status > 5

### PR DESCRIPTION
**Pull request by RangerRick**
This PR changes the SNMP tracker code to handle handle all known `error-status` codes from [RFC 3416](https://tools.ietf.org/html/rfc3416).

For the most part it treats the write-related responses as fatal (throwing an exception) but attempts retry for most other errors, in the hopes they may be transient.

While I have tested this with a few simple (mostly Linux-based) devices, I'd love a second set of eyes to not only look through the patches, but give this branch a shot and confirm I don't do anything heinous.

There is probably more refactoring that could be done to reduce duplicate code between the `AggregateTracker`, `ColumnTracker`, and `SingleInstanceTracker` classes, but I didn't attempt more rearranging for the sake of keeping the changes small.  The only change that touches more code was changing the various hardcoded codes from `static int constant`s to a single `enum` that is referenced throughout.

* JIRA: http://issues.opennms.org/browse/NMS-8671

### CODE REVIEW

**Comment by RangerRick** [comment]
Packages are testable using `branches/jira-NMS-8671` in place of `stable` your debian or yum config.

File: core/snmp/api/src/main/java/org/opennms/netmgt/snmp/ErrorStatus.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
+package org.opennms.netmgt.snmp;
```
Comment (j-white) [inline core/snmp/api/src/main/java/org/opennms/netmgt/snmp/ErrorStatus.java]: Missing copyright header.

File: core/snmp/api/src/main/java/org/opennms/netmgt/snmp/ErrorStatusException.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.opennms.netmgt.snmp;
```
Comment (j-white) [inline core/snmp/api/src/main/java/org/opennms/netmgt/snmp/ErrorStatusException.java]: Missing copyright header.

**Comment by RangerRick** [comment]
`nms01` is running this code and isn't showing any SNMP-related errors.  I say ship it! :shipit:

**Pull request merged**